clustered/client/src/main/java/org/ehcache/clustered/service/DefaultClusteringService.java

```diff
@@ -34,46 +48,113 @@
  */
 public class DefaultClusteringService implements ClusteringService {
 
-  private final ClusteringServiceConfiguration config;
+  private static final String AUTO_CREATE_QUERY = "auto-create";
+
+  private final URI clusterUri;
+  private final String entityIdentifier;
+  private final boolean autoCreate;
+
+  private Connection clusterConnection;
+  private EhcacheClientEntityFactory entityFactory;
+
+  @FindbugsSuppressWarnings("URF_UNREAD_FIELD")
+  private EhcacheClientEntity entity;
 
   public DefaultClusteringService(final ClusteringServiceConfiguration configuration) {
-    this.config = configuration;
+    URI ehcacheUri = configuration.getConnectionUrl();
+    this.clusterUri = extractClusterUri(ehcacheUri);
+    this.entityIdentifier = clusterUri.relativize(ehcacheUri).getPath();
+    this.autoCreate = AUTO_CREATE_QUERY.equalsIgnoreCase(ehcacheUri.getQuery());
+  }
+
+  private static URI extractClusterUri(URI uri) {
+    try {
+      return new URI(uri.getScheme(), uri.getAuthority(), null, null, null);
+    } catch (URISyntaxException e) {
+      throw new AssertionError(e);
+    }
   }
 
   @Override
   public void start(final ServiceProvider<Service> serviceProvider) {
-    // TODO: Implement start
+    try {
+      clusterConnection = ConnectionFactory.connect(clusterUri, new Properties());
+    } catch (ConnectionException ex) {
+      throw new RuntimeException(ex);
+    }
+    entityFactory = new EhcacheClientEntityFactory(clusterConnection);
+    if (autoCreate) {
+      try {
+        create();
+      } catch (IllegalStateException ex) {
+        //ignore - entity already exists
+      }
+    }
+    connect();
   }
 
   @Override
   public void startForMaintenance(ServiceProvider<MaintainableService> serviceProvider) {
-    // TODO: Implement maintenance
+    try {
+      clusterConnection = ConnectionFactory.connect(clusterUri, new Properties());
+    } catch (ConnectionException ex) {
+      throw new RuntimeException(ex);
+    }
+    entityFactory = new EhcacheClientEntityFactory(clusterConnection);
+    if (!entityFactory.acquireLeadership(entityIdentifier)) {
+      throw new IllegalStateException("Couldn't acquire cluster-wide maintenance lease");
+    }
   }
 
   @Override
   public void stop() {
-    // TODO: Implement stop
+    entityFactory.abandonLeadership(entityIdentifier);
```

Comment:
Yes... I believe so but I will verify this again.

- [ ] Verify it's safe to abandon leadership when not owning it (with a test maybe?)